### PR TITLE
Remove legacy plasma unlimited and pull manager pinning flag

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

These features have been stable for a few releases now; remove the old code.

### CODE REVIEW

File: src/ray/object_manager/plasma/plasma_allocator.cc [hunk 1]
```diff
@@ -73,15 +73,13 @@ absl::optional<Allocation> BuildAllocation(void *addr, size_t size) {
...
-                                 bool fallback_enabled)
+                                 bool hugepage_enabled, int64_t footprint_limit)
     : kFootprintLimit(footprint_limit),
       kAlignment(kAllocationAlignment),
-      kFallbackEnabled(fallback_enabled),
       allocated_(0),
       fallback_allocated_(0) {
   internal::SetDLMallocConfig(plasma_directory, fallback_directory, hugepage_enabled,
-                              fallback_enabled);
+                              /*fallback_enabled=*/true);
```
Comment (ericl) [inline src/ray/object_manager/plasma/plasma_allocator.cc:82]: Can remove the flag from dlmalloc as a follow-up.

**Comment by rkooo567** [comment]
cc @scv119. This will make merge conflict with https://github.com/ray-project/ray/pull/17344

**Review by rkooo567 — APPROVED** [review]

**Review by scv119 — APPROVED** [review]
image

File: src/ray/common/ray_config_def.h [hunk 2]
```diff
@@ -95,20 +95,9 @@ RAY_CONFIG(bool, lineage_pinning_enabled, false)
...
-/// Whether to never raise OOM. Instead, we fallback to allocating from the filesystem
-/// in /tmp, creating a new file per object. This degrades performance since filesystem
-/// backed objects are written to disk, but allows Ray to operate with degraded
-/// performance instead of crashing. Note that memory admission control is still in play,
-/// so Ray will still do its best to avoid running out of memory (i.e., via throttling and
-/// spilling).
-RAY_CONFIG(bool, plasma_unlimited, true)
-
 /// DEBUG-ONLY: Min number of pulls to keep active. Only supports values {0, 1}.
 RAY_CONFIG(int, pull_manager_min_active_pulls, 1)
```
Comment (stephanie-wang) [inline src/ray/common/ray_config_def.h:99]: Could we also remove this?

Comment (ericl) [reply src/ray/common/ray_config_def.h:99]: Done

**Pull request merged**